### PR TITLE
Feature/parallelize moar traversals

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PRs adds parallel traversal execution in cluster mode.

- [ ] :hankey: Bugfix
- [x] :pizza: New feature
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [x] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports (no backports)

#### Related Information

- [ ] Docs PR: 
- [x] Enterprise PR: https://github.com/arangodb/enterprise/pull/957
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/ExecutionEngine.cpp [hunk 1]
```diff
@@ -169,53 +169,61 @@ Result ExecutionEngine::createBlocks(std::vector<ExecutionNode*> const& nodes,
...
+            addBlock(std::move(r));
+          }
         }
+      } else if (en->getFirstDependency()->getType() == ExecutionNode::ASYNC) {
+        // TODO: I don't think we need to do anything here. Dependency are
+        // handled before
+        // TODO: This is just a WIP assert, make sure to delete this before
+        // merging
+        // TODO: We expect to end up here during planning.
+        // TRI_ASSERT(false);
```
Comment (hkernbach) [inline arangod/Aql/ExecutionEngine.cpp:224]: This area needs cleanup before we merge back.

**Review by hkernbach — COMMENTED** [review]
LGTM. One area needs cleanup. Will do now. Other cleanup tasks done already.

File: arangod/Aql/ExecutionEngine.cpp (hunk 1 above)
Comment (hkernbach) [reply arangod/Aql/ExecutionEngine.cpp]: Done.

**Comment by hkernbach** [comment]
LGTM (cannot approve as I'm the author - still this is co-authored with michael)

**Comment by hkernbach** [comment]
https://jenkins01.arangodb.biz/job/arangodb-matrix-pr/22326/ green
@mchacki fyi

**Review by mchacki — APPROVED** [review]
LGTM

**Pull request merged**